feat(diagnostics): Check empty tuples on LHS and missing RHS values (#1983)
NomicFoundation/slang: crates/solidity-v2/testing/snapshots/diagnostics_output/structure/empty_tuple_on_lhs/direct/generated/slang/0.8.0-failure.txt

# This file is generated automatically by infrastructure scripts. Please don't edit by hand.

Diagnostics: 1

> [structure/empty-tuple-on-lhs] Error: Empty tuple on the left hand side.

Reviewer: I know we're copying solidity here, but it'd be interesting to see if we could make these diagnostics more symmetric, ie 

```suggestion
[structure/empty-tuple-on-lhs] Error: Empty tuple on a writing position.
```

Do you think it makes sense?

Author: Hmmm... LHS or left hand side is common lingo in programming languages and compilers. In this case, "writing position" sounds weird. Also, the (somewhat) equivalent message for the RHS is "Tuple component cannot be empty" so no mention of a reading position either.

Reviewer: > equivalent message for the RHS is "Tuple component cannot be empty" so no mention of a reading position either.

Yeah, I mixed it up there somehow.

Makes sense, thanks!
